addons/hr_holidays/report/hr_leave_employee_report.py

```diff
@@ -0,0 +1,153 @@
+                    DATE_TRUNC('day', days_included_in_request) AS day_start
+                FROM hr_leave hl
+                CROSS JOIN LATERAL GENERATE_SERIES(
+                    DATE_TRUNC('day', date_from),
+                    DATE_TRUNC('day', date_to),
+                    INTERVAL '1 day'
+                ) AS days_included_in_request
+                WHERE hl.employee_company_id IN %(company_ids)s
+            )
+            SELECT
+                id, leave_id, employee_id,
+                holiday_status_id, state, description,
+                GREATEST(date_from, day_start) AS day_aligned_date_from,
+                LEAST(date_to, (day_start + INTERVAL '1 day' - INTERVAL '1 second')) AS day_aligned_date_to
+            FROM leave_data;
+            """,
+            company_ids=tuple(self.env.companies.ids),
+        ))
+        fetched_leave_field_names = [desc[0] for desc in self.env.cr.description]
+        leave_records = self.env.cr.fetchall()
+        return fetched_leave_field_names, leave_records
+
+    def _compute_is_hatched(self, report_records):
+        for report_record in report_records:
+            report_record['is_striked'] = report_record['state'] == 'refuse'
+            report_record['is_hatched'] = report_record['state'] not in ['refuse', 'validate']
+
+    def _create_report_records_from_leave_records(self, leave_records, leave_field_names):
+        report_records = [
+            {field_name: leave_record[index] for index, field_name in enumerate(leave_field_names)}
+            for leave_record in leave_records
+        ]
+        return report_records
+
+    def _compute_working_schedule_aligned_dates(self, report_records):
+        for report_record in report_records:
+            start_date = report_record['day_aligned_date_from'].replace(tzinfo=utc)
+            end_date = report_record['day_aligned_date_to'].replace(tzinfo=utc)
+            work_intervals = list(self.env['hr.leave'].browse(report_record['leave_id'])
+                .resource_calendar_id._work_intervals_batch(start_date, end_date, compute_leaves=False)[False])
+            if work_intervals:
+                report_record['working_schedule_aligned_date_from'] = work_intervals[0][0].astimezone(utc).replace(tzinfo=None)
+                report_record['working_schedule_aligned_date_to'] = work_intervals[-1][1].astimezone(utc).replace(tzinfo=None)
+            else:
+                # Fallback if no work intervals found
+                report_record['working_schedule_aligned_date_from'] = start_date.replace(tzinfo=None)
+                report_record['working_schedule_aligned_date_to'] = end_date.replace(tzinfo=None)
+
+    def _compute_leave_duration(self, report_records):
+        leave_ids = [report_record['leave_id'] for report_record in report_records]
+        leaves = self.env['hr.leave'].browse(leave_ids)
+        holiday_status_id_by_leave_id = {leave.id: leave.holiday_status_id.id for leave in leaves}
```

Comment:
I think it will be better with the `_read_group` but we will need to debug a bit to get the parameters right 😭 
```suggestion
        holiday_status_id_by_leave_id = dict(self.env['hr.leave']._read_group(
            domain=[('id', 'in', leave_ids)],
            groupby=['id'],
            aggregates=['holiday_status_id'],
        ))
```